undertow/src/main/java/org/wildfly/extension/undertow/UndertowPersistentResourceXMLDescriptionFactory.java

```diff
@@ -0,0 +1,206 @@
+        PersistentResourceXMLDescription.PersistentResourceXMLBuilder builder = builder(ServletContainerDefinition.PATH_ELEMENT);
+
+        builder.addChild(builder(JspDefinition.PATH_ELEMENT, JspDefinition.ATTRIBUTES.stream()).setXmlElementName(Constants.JSP_CONFIG));
+        builder.addChild(builder(SessionCookieDefinition.PATH_ELEMENT, SessionCookieDefinition.ATTRIBUTES.stream()));
+        builder.addChild(builder(PersistentSessionsDefinition.PATH_ELEMENT, PersistentSessionsDefinition.ATTRIBUTES.stream()));
+        builder.addChild(builder(WebsocketsDefinition.PATH_ELEMENT, WebsocketsDefinition.ATTRIBUTES.stream()));
+        builder.addChild(builder(MimeMappingDefinition.PATH_ELEMENT, MimeMappingDefinition.ATTRIBUTES.stream()).setXmlWrapperElement("mime-mappings"));
+        builder.addChild(builder(WelcomeFileDefinition.PATH_ELEMENT).setXmlWrapperElement("welcome-files"));
+        builder.addChild(builder(CrawlerSessionManagementDefinition.PATH_ELEMENT, CrawlerSessionManagementDefinition.ATTRIBUTES.stream()));
+
+        Stream<AttributeDefinition> attributes = ServletContainerDefinition.ATTRIBUTES.stream();
+        attributes.forEach(builder::addAttribute);
+        return builder;
+    }
+
+    private static PersistentResourceXMLDescription.PersistentResourceXMLBuilder modClusterBuilder(UndertowSchema schema) {
+        PersistentResourceXMLDescription.PersistentResourceXMLBuilder builder = builder(ModClusterDefinition.PATH_ELEMENT);
+
+        builder.addChild(builder(NoAffinityResourceDefinition.PATH).setXmlElementName(Constants.NO_AFFINITY));
+        builder.addChild(builder(SingleAffinityResourceDefinition.PATH).setXmlElementName(Constants.SINGLE_AFFINITY));
+        builder.addChild(builder(RankedAffinityResourceDefinition.PATH, Attribute.stream(RankedAffinityResourceDefinition.Attribute.class)).setXmlElementName(Constants.RANKED_AFFINITY));
+
+        Stream<AttributeDefinition> attributes = ModClusterDefinition.ATTRIBUTES.stream();
+        attributes.forEach(builder::addAttribute);
+        return builder;
+    }
+
+    private static PersistentResourceXMLDescription.PersistentResourceXMLBuilder applicationSecurityDomainBuilder(UndertowSchema schema) {
+        PersistentResourceXMLDescription.PersistentResourceXMLBuilder builder = builder(ApplicationSecurityDomainDefinition.PATH_ELEMENT).setXmlWrapperElement(Constants.APPLICATION_SECURITY_DOMAINS);
+
+        Stream<AttributeDefinition> ssoAttributes = Stream.concat(Attribute.stream(ApplicationSecurityDomainSingleSignOnDefinition.Attribute.class), Attribute.stream(SingleSignOnDefinition.Attribute.class));
+        builder.addChild(builder(SingleSignOnDefinition.PATH_ELEMENT, ssoAttributes));
+
+        Stream<AttributeDefinition> attributes = ApplicationSecurityDomainDefinition.ATTRIBUTES.stream();
+        attributes.forEach(builder::addAttribute);
+        return builder;
+    }
+
+    // TODO Drop methods below once WFCORE-6218 is available
```

Comment:
I see that the PR for WFCORE-6218 is already reviewed: https://github.com/wildfly/wildfly-core/pull/5369 so hopefully we will be able to drop these pretty soon. Should I create a follow up Jira, or should we assign this to WFLY-14877?

Reply:
I created https://issues.redhat.com/browse/WFLY-17611 to track this.  I can handle that cleanup once the requisite wildfly-core version is upgraded.